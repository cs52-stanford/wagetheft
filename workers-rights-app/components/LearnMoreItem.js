import React from "react";
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Image,
  Platform,
  TouchableNativeFeedback,
} from "react-native";
import PropTypes from "prop-types";
import Colors from "../constants/Colors";
import ImportedData from "../data/FetchRightsData"; //eslint-disable-line

const LearnMoreItem = ({ id, onPress }) => {
  const selectedLearnMore = ImportedData.getLearnMores().find(
    (learnmore) => learnmore.id === id
  );
  if (!selectedLearnMore) {
    return null;
  }

  const TouchablePlatformSpecific =
    Platform.OS === "ios" ? TouchableOpacity : TouchableNativeFeedback;

  return (
    <TouchablePlatformSpecific onPress={onPress}>
      <View style={styles.container}>
        <View style={styles.imageContainer}>
          <Image
            source={require("../images/question.png")}
            style={{ width: "100%", height: "100%" }}
          />
        </View>
        <Text numberOfLines={2} style={styles.subrightstxt}>
          {selectedLearnMore.title}
        </Text>
      </View>
    </TouchablePlatformSpecific>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    width: "99%",
    height: 60,
    marginHorizontal: "0.5%",
    marginVertical: 5,
    paddingLeft: 10,
    paddingRight: 5,
    borderRadius: 8,
    backgroundColor: Colors.lightOrange,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.2,
    shadowRadius: 1.41,
    elevation: 2,
  },

  imageContainer: {
    width: 40,
    height: 40,
    marginRight: 10,
    alignItems: "center",
    justifyContent: "center",
  },
  subrightstxt: {
    fontSize: 14,
    flex: 1,
  },
});

LearnMoreItem.propTypes = {
  id: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

export default LearnMoreItem;
