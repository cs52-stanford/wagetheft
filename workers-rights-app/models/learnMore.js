class learnMore {
  constructor(id, title, image, informationChunks) {
    this.id = id;
    this.title = title;
    this.image = image;
    this.informationChunks = informationChunks;
  }
}

export default learnMore;
