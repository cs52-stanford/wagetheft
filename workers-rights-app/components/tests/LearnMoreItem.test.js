import React from "react";
import ImportedData from "../../data/FetchRightsData";
import LearnMoreItem from "../LearnMoreItem";
import learnMore from "../../models/learnMore";
import { render, fireEvent } from "@testing-library/react-native";

jest.mock("../../data/FetchRightsData");

describe("<LearnMoreItem /> Tests", () => {
  it("should match snapshot", () => {
    const mockPressFunction = jest.fn();
    const mockGetLearnMores = jest.fn();
    const mockLearnMore = new learnMore("DummyID", "Dummy Title", [
      { header: "Dummy Header", body: "Dummy Body" },
    ]);
    mockGetLearnMores.mockReturnValue([mockLearnMore]);

    ImportedData.getLearnMores = mockGetLearnMores;
    const rendered = render(
      <LearnMoreItem id="DummyID" onPress={mockPressFunction} />
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it("onPress should be called", () => {
    const mockPressFunction = jest.fn();
    const mockGetLearnMores = jest.fn();
    const mockLearnMore = new learnMore("DummyID", "Dummy Title", [
      { header: "Dummy Header", body: "Dummy Body" },
    ]);
    mockGetLearnMores.mockReturnValue([mockLearnMore]);

    ImportedData.getLearnMores = mockGetLearnMores;
    const rendered = render(
      <LearnMoreItem id="DummyID" onPress={mockPressFunction} />
    );

    const learnMoreItem = rendered.getByTestId("learnMore");
    fireEvent(learnMoreItem, "press");
    expect(mockPressFunction).toHaveBeenCalled();
  });

  it("text should be displayed properly", () => {
    const mockPressFunction = jest.fn();
    const mockGetLearnMores = jest.fn();
    const mockLearnMore = new learnMore("DummyID", "Dummy Title", [
      { header: "Dummy Header", body: "Dummy Body" },
    ]);
    mockGetLearnMores.mockReturnValue([mockLearnMore]);

    ImportedData.getLearnMores = mockGetLearnMores;
    const rendered = render(
      <LearnMoreItem id="DummyID" onPress={mockPressFunction} />
    );

    const textComponent = rendered.getByTestId("text");
    expect(textComponent.props.children).toEqual("Dummy Title");
  });
});
