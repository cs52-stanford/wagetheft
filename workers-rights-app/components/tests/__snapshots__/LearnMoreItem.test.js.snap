// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LearnMoreItem /> Tests should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="learnMore"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF6EB",
        "borderRadius": 8,
        "elevation": 2,
        "flexDirection": "row",
        "height": 60,
        "marginHorizontal": "0.5%",
        "marginVertical": 5,
        "paddingLeft": 10,
        "paddingRight": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 1.41,
        "width": "99%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginRight": 10,
          "width": 40,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
    <Text
      numberOfLines={2}
      style={
        Object {
          "flex": 1,
          "fontSize": 14,
        }
      }
      testID="text"
    >
      Dummy Title
    </Text>
  </View>
</View>
`;
